### PR TITLE
feat: update device OBM info in Ironic

### PR DESCRIPTION
**Pull request by cardoe**
Ensure that the OBM credentials are provided to Ironic when we update
the device based on having an iDRAC or iLO interface. The Ironic API
doesn't do 'replace' operations so some extra care must be made to
detect if the values are currently set, especially for values that
cannot be retrieved like the password, to ensure we do an 'add'
operation when needed and a 'replace' when needed.

### CODE REVIEW

**Comment by andrrax** [comment]
The argo-workflow stuff looks good. We should probably bump this though:

https://github.com/rackerlabs/understack/blob/main/.github/workflows/build-container-images.yaml#L23

File: argo-workflows/ironic-nautobot-sync/workflowtemplates/synchronize-obm-creds.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+              name: get-obm-creds
+              templateRef: main
+            arguments:
+              - name: hostname
+                value: "{{ workflow.parameters.interface_update_event.data.device.name }}"
+        - - name: synchronize-obm-creds
+            template: synchronize-obm-creds
+            arguments:
+              - name: obm
+                value: "{{ steps.load-obm-creds.outputs.secret }}"
```
Comment (andrrax) [inline argo-workflows/ironic-nautobot-sync/workflowtemplates/synchronize-obm-creds.yaml:25]:
```suggestion
        - - name: load-obm-creds
            templateRef:
              name: get-obm-creds
              template: main
            arguments:
              parameters:
                - name: hostname
                  value: "{{ workflow.parameters.interface_update_event.data.device.name }}"
        - - name: synchronize-obm-creds
            template: synchronize-obm-creds
            arguments:
              parameters:
                - name: obm
                  value: "{{ steps.load-obm-creds.outputs.secret }}"
```

**Comment by skrobul** [comment]
Added a commit to handle unset passwords and bring this more in sync with #114

**Comment by skrobul** [comment]
```
MESSAGE
task 'synchronize-obm-creds-dj7qn[0].load-obm-creds.get-obm-creds.get-obm-creds-core.get-core-num' errored: invalid cache key: {{ jsonpath(workflow-parameters-interface_update_event-json, '$-data-device-name') }}
```

image

**Review by nicholaskuechler — APPROVED** [review]

**Pull request merged**